pipeline-model-definition/src/test/resources/whenBranchFalse.groovy

```diff
@@ -24,6 +24,9 @@
 
 pipeline {
     agent any
+    environment {
+        BRANCH_NAME = "master"
```

Comment:
Then perhaps we should be looking for some other way of figuring out what branch it is. But that could be a future enhancement.